src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/AspNetCoreMvcDiagnosticsListener.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DiagnosticAdapter;
+
+namespace Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners
+{
+    public class AspNetCoreMvcDiagnosticsListener: IApplicationInsightDiagnosticListener
+    {
+        public const string BeforeActionNotificationName = "Microsoft.AspNetCore.Mvc.BeforeAction";
+
+        public string ListenerName { get; } = "Microsoft.AspNetCore";
+
+        [DiagnosticName(BeforeActionNotificationName)]
+        public void OnBeforeAction(HttpContext httpContext, IRouteData routeData)
+        {
+            var telemetry = httpContext.Features.Get<RequestTelemetry>();
+
+            if (telemetry != null && string.IsNullOrEmpty(telemetry.Name))
+            {
+                string name = this.GetNameFromRouteContext(routeData);
+
+                if (!string.IsNullOrEmpty(name))
+                {
+                    name = httpContext.Request.Method + " " + name;
+                    telemetry.Name = name;
```

Comment:
@SergeyKanzhelev Does this value for name match with the schema changes discussed in [https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/202](https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/202)?